blockchain/blocks/src/tipset.rs

```diff
@@ -173,14 +186,9 @@ impl FullTipset {
     // TODO: conversions from full to regular tipset should not return a result
     // and should be validated on creation instead
     /// Returns a Tipset
-    pub fn into_tipset(self) -> Result<Tipset, Error> {
-        let headers = self.blocks.into_iter().map(|block| block.header).collect();
-        Tipset::new(headers)
-    }
-    /// Returns a Tipset
-    pub fn to_tipset(&self) -> Result<Tipset, Error> {
+    pub fn to_tipset(&self) -> Tipset {
         let headers = self.blocks.iter().map(Block::header).cloned().collect();
-        Tipset::new(headers)
+        Tipset::new(headers).unwrap()
```

Comment:
using unwrap here opens us up to a panic, the validation should be the same at this point so you should be able to construct the struct without going through the `::new(..)` constructor. You may need to abstract some logic around sorting (or ensuring the blocks are sorted) in the `FullTipset` constructor to make sure this is never inconsistent.

Reply:
Just pushed a new commit that sorts the blocks in the `FullTipset` constructor and removed calls to the `Tipset` constructor when converting from `FullTipset` to `Tipset`.